### DISCUSSION TITLE
Edge classification

### DISCUSSION THREAD

**Discussion by shrimonmuke0202** [pyg-team/pytorch_geometric#4205] [Q&A]
How can I apply GCN for edge classification tasks using Pytorch Geometric? and the classification is multiclass classification.

**Comment by rusty1s** [comment]
This is possible, and involves using `GCN` to learn node embeddings which can then be used to learn edge-level embeddings, e.g.:
```python
model = GCN(in_channels, hidden_channels, num_layers)
mlp = MLP(2 * hidden_channels, num_classes, num_layers)

h = model(x, edge_index)

h_src = h[edge_index[0]]
h_dst = h[edge_index[1]]

return mlp(torch.cat([h_src, h_dst], dim=-1))
```
